File: en/java-spring-webflux/src/main/java/com/example/controller/PaymentController.java
package com.example.controller;

import org.springframework.web.bind.annotation.*;
import com.example.model.Payment;
import com.example.dto.Result;
import com.example.dto.PageResult;
import jakarta.validation.Valid;
import java.math.BigDecimal;
import java.util.List;

@RestController
@RequestMapping("/api/payments")
public class PaymentController {
    
    @GetMapping
    public Result<PageResult<Payment>> listPayments(
            @RequestParam(required = false) Long orderId,
            @RequestParam(required = false) String paymentNo,
            @RequestParam(required = false) Integer status,
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @PostMapping
    public Result<Payment> createPayment(@Valid @RequestBody CreatePaymentReq req) {
        return Result.success(new Payment());
    }
    
    @GetMapping("/{id}")
    public Result<Payment> getPayment(@PathVariable Long id) {
        return Result.success(new Payment());
    }
    
    @GetMapping("/no/{paymentNo}")
    public Result<Payment> getPaymentByNo(@PathVariable String paymentNo) {
        return Result.success(new Payment());
    }
    
    @PostMapping("/{id}/cancel")
    public Result<Payment> cancelPayment(@PathVariable Long id) {
        return Result.success(new Payment());
    }
    
    @PostMapping("/{id}/refund")
    public Result<RefundVO> processRefund(@PathVariable Long id, @Valid @RequestBody RefundReq req) {
        return Result.success(new RefundVO());
    }
    
    @GetMapping("/order/{orderId}")
    public Result<List<Payment>> getPaymentsByOrder(@PathVariable Long orderId) {
        return Result.success(List.of());
    }
    
    @GetMapping("/history")
    public Result<PageResult<Payment>> getPaymentHistory(
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate,
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @PostMapping("/{id}/verify")
    public Result<PaymentVerifyVO> verifyPayment(@PathVariable Long id, @RequestParam String transactionId) {
        return Result.success(new PaymentVerifyVO());
    }
    
    @GetMapping("/methods")
    public Result<List<PaymentMethodVO>> getPaymentMethods() {
        return Result.success(List.of());
    }
    
    @GetMapping("/statistics")
    public Result<PaymentStatisticsVO> getStatistics(
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate) {
        return Result.success(new PaymentStatisticsVO());
    }
    
    @GetMapping("/pending")
    public Result<PageResult<Payment>> getPendingPayments(
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @GetMapping("/failed")
    public Result<PageResult<Payment>> getFailedPayments(
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @PostMapping("/{id}/retry")
    public Result<Payment> retryPayment(@PathVariable Long id) {
        return Result.success(new Payment());
    }
    
    @GetMapping("/{id}/status")
    public Result<PaymentStatusVO> getPaymentStatus(@PathVariable Long id) {
        return Result.success(new PaymentStatusVO());
    }
    
    @PostMapping("/callback")
    public Result<Void> paymentCallback(@RequestBody String callbackData) {
        return Result.success(null);
    }
    
    @GetMapping("/export")
    public Result<String> exportPayments(
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate) {
        return Result.success("export_file_url");
    }
    
    @GetMapping("/{id}/receipt")
    public Result<PaymentReceiptVO> getReceipt(@PathVariable Long id) {
        return Result.success(new PaymentReceiptVO());
    }
    
    @PostMapping("/{id}/send-receipt")
    public Result<Void> sendReceipt(@PathVariable Long id, @RequestParam String email) {
        return Result.success(null);
    }
}

class CreatePaymentReq {
    private Long orderId;
    private BigDecimal amount;
    private String paymentMethod;
    
    public Long getOrderId() { return orderId; }
    public void setOrderId(Long orderId) { this.orderId = orderId; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(String paymentMethod) { this.paymentMethod = paymentMethod; }
}

class RefundReq {
    private BigDecimal amount;
    private String reason;
    
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }
}

class RefundVO {
    private Long id;
    private String refundNo;
    private Long paymentId;
    private BigDecimal amount;
    private String reason;
    private Integer status;
    private String processedAt;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getRefundNo() { return refundNo; }
    public void setRefundNo(String refundNo) { this.refundNo = refundNo; }
    public Long getPaymentId() { return paymentId; }
    public void setPaymentId(Long paymentId) { this.paymentId = paymentId; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getProcessedAt() { return processedAt; }
    public void setProcessedAt(String processedAt) { this.processedAt = processedAt; }
}

class PaymentVerifyVO {
    private Long paymentId;
    private Boolean verified;
    private String transactionId;
    private String verifiedAt;
    
    public Long getPaymentId() { return paymentId; }
    public void setPaymentId(Long paymentId) { this.paymentId = paymentId; }
    public Boolean getVerified() { return verified; }
    public void setVerified(Boolean verified) { this.verified = verified; }
    public String getTransactionId() { return transactionId; }
    public void setTransactionId(String transactionId) { this.transactionId = transactionId; }
    public String getVerifiedAt() { return verifiedAt; }
    public void setVerifiedAt(String verifiedAt) { this.verifiedAt = verifiedAt; }
}

class PaymentMethodVO {
    private String code;
    private String name;
    private String icon;
    private Boolean enabled;
    
    public String getCode() { return code; }
    public void setCode(String code) { this.code = code; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getIcon() { return icon; }
    public void setIcon(String icon) { this.icon = icon; }
    public Boolean getEnabled() { return enabled; }
    public void setEnabled(Boolean enabled) { this.enabled = enabled; }
}

class PaymentStatisticsVO {
    private Long totalPayments;
    private Long successfulPayments;
    private Long failedPayments;
    private BigDecimal totalAmount;
    private BigDecimal refundedAmount;
    
    public Long getTotalPayments() { return totalPayments; }
    public void setTotalPayments(Long totalPayments) { this.totalPayments = totalPayments; }
    public Long getSuccessfulPayments() { return successfulPayments; }
    public void setSuccessfulPayments(Long successfulPayments) { this.successfulPayments = successfulPayments; }
    public Long getFailedPayments() { return failedPayments; }
    public void setFailedPayments(Long failedPayments) { this.failedPayments = failedPayments; }
    public BigDecimal getTotalAmount() { return totalAmount; }
    public void setTotalAmount(BigDecimal totalAmount) { this.totalAmount = totalAmount; }
    public BigDecimal getRefundedAmount() { return refundedAmount; }
    public void setRefundedAmount(BigDecimal refundedAmount) { this.refundedAmount = refundedAmount; }
}

class PaymentStatusVO {
    private Long paymentId;
    private String paymentNo;
    private Integer status;
    private String statusDescription;
    private String updatedAt;
    
    public Long getPaymentId() { return paymentId; }
    public void setPaymentId(Long paymentId) { this.paymentId = paymentId; }
    public String getPaymentNo() { return paymentNo; }
    public void setPaymentNo(String paymentNo) { this.paymentNo = paymentNo; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getStatusDescription() { return statusDescription; }
    public void setStatusDescription(String statusDescription) { this.statusDescription = statusDescription; }
    public String getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
}

class PaymentReceiptVO {
    private String receiptNo;
    private String paymentNo;
    private BigDecimal amount;
    private String paymentMethod;
    private String paidAt;
    private String receiptUrl;
    
    public String getReceiptNo() { return receiptNo; }
    public void setReceiptNo(String receiptNo) { this.receiptNo = receiptNo; }
    public String getPaymentNo() { return paymentNo; }
    public void setPaymentNo(String paymentNo) { this.paymentNo = paymentNo; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(String paymentMethod) { this.paymentMethod = paymentMethod; }
    public String getPaidAt() { return paidAt; }
    public void setPaidAt(String paidAt) { this.paidAt = paidAt; }
    public String getReceiptUrl() { return receiptUrl; }
    public void setReceiptUrl(String receiptUrl) { this.receiptUrl = receiptUrl; }
}

File: en/kotlin-springmvc/src/main/kotlin/com/example/controller/PaymentController.kt
package com.example.controller

import com.example.dto.PageResult
import com.example.dto.Result
import com.example.model.Payment
import org.springframework.web.bind.annotation.*
import java.math.BigDecimal

@RestController
@RequestMapping("/api/payments")
class PaymentController {
    
    @GetMapping
    fun listPayments(
        @RequestParam(required = false) orderId: Long?,
        @RequestParam(required = false) paymentNo: String?,
        @RequestParam(required = false) status: Int?,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Payment>> {
        val result = PageResult<Payment>()
        return Result.success(result)
    }
    
    @PostMapping
    fun createPayment(@RequestBody req: CreatePaymentReq): Result<Payment> {
        val payment = Payment()
        return Result.success(payment)
    }
    
    @GetMapping("/{id}")
    fun getPayment(@PathVariable id: Long): Result<Payment> {
        val payment = Payment()
        return Result.success(payment)
    }
    
    @GetMapping("/no/{paymentNo}")
    fun getPaymentByNo(@PathVariable paymentNo: String): Result<Payment> {
        val payment = Payment()
        return Result.success(payment)
    }
    
    @PostMapping("/{id}/cancel")
    fun cancelPayment(@PathVariable id: Long): Result<Payment> {
        val payment = Payment()
        return Result.success(payment)
    }
    
    @PostMapping("/{id}/refund")
    fun processRefund(@PathVariable id: Long, @RequestBody req: RefundReq): Result<RefundVO> {
        val refund = RefundVO()
        return Result.success(refund)
    }
    
    @GetMapping("/order/{orderId}")
    fun getPaymentsByOrder(@PathVariable orderId: Long): Result<List<Payment>> {
        val payments = listOf<Payment>()
        return Result.success(payments)
    }
    
    @GetMapping("/history")
    fun getPaymentHistory(
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Payment>> {
        val result = PageResult<Payment>()
        return Result.success(result)
    }
    
    @PostMapping("/{id}/verify")
    fun verifyPayment(@PathVariable id: Long, @RequestParam transactionId: String): Result<PaymentVerifyVO> {
        val verify = PaymentVerifyVO()
        return Result.success(verify)
    }
    
    @GetMapping("/methods")
    fun getPaymentMethods(): Result<List<PaymentMethodVO>> {
        val methods = listOf<PaymentMethodVO>()
        return Result.success(methods)
    }
    
    @GetMapping("/statistics")
    fun getStatistics(
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?
    ): Result<PaymentStatisticsVO> {
        val statistics = PaymentStatisticsVO()
        return Result.success(statistics)
    }
    
    @GetMapping("/pending")
    fun getPendingPayments(
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Payment>> {
        val result = PageResult<Payment>()
        return Result.success(result)
    }
    
    @GetMapping("/failed")
    fun getFailedPayments(
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Payment>> {
        val result = PageResult<Payment>()
        return Result.success(result)
    }
    
    @PostMapping("/{id}/retry")
    fun retryPayment(@PathVariable id: Long): Result<Payment> {
        val payment = Payment()
        return Result.success(payment)
    }
    
    @GetMapping("/{id}/status")
    fun getPaymentStatus(@PathVariable id: Long): Result<PaymentStatusVO> {
        val status = PaymentStatusVO()
        return Result.success(status)
    }
    
    @PostMapping("/callback")
    fun paymentCallback(@RequestBody callbackData: String): Result<Void> {
        return Result.success(null)
    }
    
    @GetMapping("/export")
    fun exportPayments(
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?
    ): Result<String> {
        val fileUrl = ""
        return Result.success(fileUrl)
    }
    
    @GetMapping("/{id}/receipt")
    fun getReceipt(@PathVariable id: Long): Result<PaymentReceiptVO> {
        val receipt = PaymentReceiptVO()
        return Result.success(receipt)
    }
    
    @PostMapping("/{id}/send-receipt")
    fun sendReceipt(@PathVariable id: Long, @RequestParam email: String): Result<Void> {
        return Result.success(null)
    }
}

data class CreatePaymentReq(
    val orderId: Long?,
    val amount: BigDecimal?,
    val paymentMethod: String?
)

data class RefundReq(
    val amount: BigDecimal?,
    val reason: String?
)

data class RefundVO(
    val id: Long? = null,
    val refundNo: String? = null,
    val paymentId: Long? = null,
    val amount: BigDecimal? = null,
    val reason: String? = null,
    val status: Int? = null,
    val processedAt: String? = null
)

data class PaymentVerifyVO(
    val paymentId: Long? = null,
    val verified: Boolean? = null,
    val transactionId: String? = null,
    val verifiedAt: String? = null
)

data class PaymentMethodVO(
    val code: String? = null,
    val name: String? = null,
    val icon: String? = null,
    val enabled: Boolean? = null
)

data class PaymentStatisticsVO(
    val totalPayments: Long? = null,
    val successfulPayments: Long? = null,
    val failedPayments: Long? = null,
    val totalAmount: BigDecimal? = null,
    val refundedAmount: BigDecimal? = null
)

data class PaymentStatusVO(
    val paymentId: Long? = null,
    val paymentNo: String? = null,
    val status: Int? = null,
    val statusDescription: String? = null,
    val updatedAt: String? = null
)

data class PaymentReceiptVO(
    val receiptNo: String? = null,
    val paymentNo: String? = null,
    val amount: BigDecimal? = null,
    val paymentMethod: String? = null,
    val paidAt: String? = null,
    val receiptUrl: String? = null
)
